sagemathinc/cocalc
# phase 2: removing landing pages

# Description

included in this package
* removes most of our landing pages, they're in the hub-landing plugin now
* adds a bit more customization
* replaces the index page with a shorter page, which is a lot more customizable
* enhances the signin-landing page (logo, banner, etc.)
* tweaks to the about page (logo, banner, name, support contact, etc.)
* footer
* there are also changes to the initial loading banner. i.e. it's as usual if there is nothing configured. otherwise, it spells out the site name, support email address, and of course, the images

more general
* what's interesting: I see new typescript warnings. not sure why they pop up. I'll fix them as well.
* once this is merged, hub-landing requires a small patch as well

## rollout
1. hub first
2. then static

# Testing Steps
* **in any case, testing this requires to check this thoroughly in "test". I've not done that yet.**
* opening all application pages
* sign in works, including password reset
* index page, also to check that elements like the terms of service link only show up, if there is a ToS url set
* relevant for cocalc.com: removing all site settings except details like help email and the terms of service information string, should not change anything (loading banner, about, signin landing page...)


# Screenshots

index page

![screenshot-cocalc com-2020 03 23-14_47_32](https://user-images.githubusercontent.com/207405/77323513-acdb1080-6d15-11ea-8a00-2db6817095ca.png)

about page

![Screenshot from 2020-03-23 14-50-02](https://user-images.githubusercontent.com/207405/77323533-b4021e80-6d15-11ea-8cd0-449911a9b9ad.png)

loading banner

![Screenshot from 2020-03-23 14-51-15](https://user-images.githubusercontent.com/207405/77323707-f9bee700-6d15-11ea-9cc9-e08a9b548126.png)


![Screenshot from 2020-03-23 14-52-12](https://user-images.githubusercontent.com/207405/77323705-f88dba00-6d15-11ea-8064-cb1eb10f7e5c.png)

landing page (pale blue background is necessary to avoid having to define 4 images, instead of 2)

![Screenshot from 2020-03-23 14-53-18](https://user-images.githubusercontent.com/207405/77323850-2d9a0c80-6d16-11ea-9200-7cfc9c5e05da.png)




### [Checklist](https://github.com/sagemathinc/cocalc/wiki/PR-Checklist):

Front end:
